[DO NOT MERGE] Calculate based off portions and incorporate dust

Doy-lee opened the pull request:
This is a regression introduced when we started calculating minimum contribution in Loki and not accounting for dust in the relaxed contribution rules.

Edit
Not to be merged until hard fork 12. Use this as a reference to fix up the problem at the source. The min portions for node calculation will switch to portions for hf12, hf11 should continue calculating off the loki amount to avoid any discrepancies.

jagerman commented:
I tried out this PR and ran into some serious issues.  It *did* work to generate a successful registration, but then it failed to pick up the SN registration from the chain (though did claim to be synced).  I stopped it, popped 500 blocks, and let it resync and ran into consensus failure.  I get stuck with:

```
Mar 25 21:27:14 keynes lokid[2266140]: 2019-03-26 00:27:14.891	[P2P8]	INFO	global	src/cryptonote_protocol/cryptonote_protocol_handler.inl:348	[65.50.230.30:22022 OUT] Sync data returned a new top block candidate: 234948 -> 235032 [Your node is 84 blocks (0 days) behind]
Mar 25 21:27:14 keynes lokid[2266140]: SYNCHRONIZATION started
Mar 25 21:27:15 keynes lokid[2266140]: 2019-03-26 00:27:15.805	[P2P8]	ERROR	service_nodes	src/cryptonote_core/service_node_list.cpp:1392	Service node reward winner is incorrect
Mar 25 21:27:15 keynes lokid[2266140]: 2019-03-26 00:27:15.825	[P2P0]	INFO	global	src/cryptonote_protocol/cryptonote_protocol_handler.inl:348	[159.89.199.190:22022 OUT] Sync data returned a new top block candidate: 234948 -> 235032 [Your node is 84 blocks (0 days) behind]
Mar 25 21:27:15 keynes lokid[2266140]: SYNCHRONIZATION started
Mar 25 21:27:16 keynes lokid[2266140]: 2019-03-26 00:27:16.344	[P2P3]	ERROR	service_nodes	src/cryptonote_core/service_node_list.cpp:1392	Service node reward winner is incorrect
Mar 25 21:27:17 keynes lokid[2266140]: 2019-03-26 00:27:17.225	[P2P9]	ERROR	service_nodes	src/cryptonote_core/service_node_list.cpp:1392	Service node reward winner is incorrect
Mar 25 21:27:17 keynes lokid[2266140]: 2019-03-26 00:27:17.335	[P2P7]	ERROR	service_nodes	src/cryptonote_core/service_node_list.cpp:1392	Service node reward winner is incorrect
Mar 25 21:27:18 keynes lokid[2266140]: 2019-03-26 00:27:18.860	[P2P9]	ERROR	service_nodes	src/cryptonote_core/service_node_list.cpp:1392	Service node reward winner is incorrect
Mar 25 21:27:19 keynes lokid[2266140]: 2019-03-26 00:27:19.162	[P2P8]	ERROR	service_nodes	src/cryptonote_core/service_node_list.cpp:1392	Service node reward winner is incorrect
Mar 25 21:27:19 keynes lokid[2266140]: 2019-03-26 00:27:19.756	[P2P7]	ERROR	service_nodes	src/cryptonote_core/service_node_list.cpp:1392	Service node reward winner is incorrect
Mar 25 21:27:19 keynes lokid[2266140]: 2019-03-26 00:27:19.885	[P2P0]	ERROR	service_nodes	src/cryptonote_core/service_node_list.cpp:1392	Service node reward winner is incorrect
Mar 25 21:27:20 keynes lokid[2266140]: 2019-03-26 00:27:20.048	[P2P3]	ERROR	service_nodes	src/cryptonote_core/service_node_list.cpp:1392	Service node reward winner is incorrect
Mar 25 21:27:20 keynes lokid[2266140]: 2019-03-26 00:27:20.690	[P2P3]	ERROR	service_nodes	src/cryptonote_core/service_node_list.cpp:1392	Service node reward winner is incorrect
```

I also tried this starting from an lmdb snapshot from last night; same result.

I reverted it, recompiled, and synced perfectly (including picking up the SN registration tx).

Doy-lee commented:
Yeah this is super not ready. There's still a bug somewhere in it, hasn't been passing my integration tests yet so I haven't addressed the review yet.

jagerman commented:
The current PR seems to work for me -- have tested it out with a few registrations of fully closed nodes, and all have gone through perfectly as long as both the wallet and node that does the `prepare_registration` are patched.  Regular 3.0.2 nodes are perfectly happy with the resulting registrations.

jagerman commented:
Maybe for HF12 the portions code should be scrapped completely: keep everything in terms of LOKI (as issue #478.  A registration would contain the fork height at the time it was generated, which would exactly define the required amounts in LOKI, and these would not change even if submitted or mined 10 blocks later.

AFAIK the point of expressing everything in portions was just to allow auto-renewal of shared contributor nodes to renew with exactly the same portions, but that isn't relevant anymore.
